vernac/vernacentries.ml

```diff
@@ -1054,10 +1058,18 @@ let vernac_end_section {CAst.loc; v} =
 let vernac_name_sec_hyp {v=id} set = Proof_using.name_set id set
 
 (* Dispatcher of the "End" command *)
+let msg_of_subsection ss =
+  match ss with
+  | Lib.OpenedModule (false,_,_,_) -> "module"
+  | Lib.OpenedModule (true,_,_,_) -> "module type"
+  | Lib.OpenedSection _ -> "section"
+  | _ -> "unknown"
 
 let vernac_end_segment ({v=id} as lid) =
-  DeclareObl.check_can_close lid.v;
-  match Lib.find_opening_node id with
+  let ss = Lib.find_opening_node id in
+  let msg = msg_of_subsection ss in
+  DeclareObl.check_solved_obligations ~msg;
```

Comment:
Why do we change printing from `M has unsolved obligations" to "module has unsolved obligations" here?
Also calling the argument `msg` sounds like it is the whole or at least main part of the message but it's actually a small bit.
Also we may want to pass the location? Not sure.

Reply:
Let me see, I remember having a conflict here with other PR so maybe I didn't resolve it properly?

Reply:
I think it was just an oversight, I dunno why I did it this way, specially when I was aware of it O_O Fixed, thanks.

Reply:
I still don't like calling it `msg`, maybe `what_for` works for you?

Reply:
Amended.